File: 03. Programming - C#/TestPreparation4Example/Triangle.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace TestPreparation4Example
{
    class Triangle:Shape
    {
        private double a;
        public double A
        {
            get
            {
                return a;
            }
            set
            {
                a = value;
            }
        }

        private double b;
        public double B
        {
            get
            {
                return b;
            }
            set
            {
                b = value;
            }
        }
        private double c;
        public double C
        {
            get
            {
                return c;
            }set => c = value;
        }

        private double ha;
        public double Ha
        {
            get
            {
                return ha;
            }set => ha = value;
        }

        public Triangle(string name, int numberOfSides,string type,double a,double b, double c, double ha)
            :base(name,numberOfSides,type)
        {
            this.a = a;
            this.b = b;
            this.c = c;
            this.ha = ha;
        }

        public override double calculateArea()
        {
            return (a * ha) / 2;
        }

        public override double calculatePerimeter()
        {
            return a + b + c;
        }
    }
}

File: 02. Programming - Java/Lesson7/src/uni/fmi/bachelors/Triangle.java
package uni.fmi.bachelors;

public class Triangle extends Shape {

	private double a;
	private double b;
	private double c;
	private double ha;
	
	public Triangle(String type, int numberOfSides, double a, double b, double c, double ha) {
		super(type, numberOfSides);
		this.a=a;
		this.b=b;
		this.c=c;
		this.ha=ha;
	}

	@Override
	public double calculatePerimeter() {
		return a+b+c;
	}

	@Override
	public double calculateArea() {
		return a*ha/2;
	}

	public double getA() {
		return a;
	}

	public void setA(double a) {
		this.a = a;
	}

	public double getB() {
		return b;
	}

	public void setB(double b) {
		this.b = b;
	}

	public double getC() {
		return c;
	}

	public void setC(double c) {
		this.c = c;
	}

	public double getHa() {
		return ha;
	}

	public void setHa(double ha) {
		this.ha = ha;
	}

	
}
